packages/core/src/Cardano/types/Cip1854PublicKey.ts

```diff
@@ -0,0 +1,32 @@
+import * as BaseEncoding from '@scure/base';
+import { InvalidStringError, OpaqueString, assertIsBech32WithPrefix, assertIsHexString } from '@cardano-sdk/util';
+
+const MAX_BECH32_LENGTH_LIMIT = 1023;
+export type Cip1854PublicKey = OpaqueString<'Cip1854PublicKey'>;
+
+export const Cip1854PublicKey = (value: string): Cip1854PublicKey => {
+  try {
+    assertIsBech32WithPrefix(value, ['acct_shared_xvk']);
+  } catch {
+    throw new InvalidStringError(value, 'Expected key to be a bech32 encoded string');
+  }
+
+  return value as Cip1854PublicKey;
+};
+
+Cip1854PublicKey.toBech32 = (value: string): Cip1854PublicKey => {
```

Comment:
Don't need to validate length then as @AngelCastilloB requested, since `Bip32PublicKeyHex` type already does it
```suggestion
Cip1854PublicKey.fromBip32PublicKeyHex = (value: Bip32PublicKeyHex): Cip1854PublicKey => {
```

Reply:
Fixed on c66a9c0